decidim-core/lib/decidim/commands/destroy_resource.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Commands
+    class DestroyResource < ::Decidim::Command
+      # Initializes the command.
+      #
+      # @param resource [ActiveRecord::Base] the resource to destroy.
+      # @param current_user [Decidim::User] the current user.
+      def initialize(resource, current_user)
+        @resource = resource
+        @current_user = current_user
+      end
+
+      # Destroys the result.
+      #
+      # Broadcasts :ok if successful, :invalid otherwise.
+      def call
+        return broadcast(:invalid) if invalid?
+
+        run_before_hooks
+        destroy_resource
+        run_after_hooks
+
+        broadcast(:ok, resource)
+      rescue Decidim::Commands::HookError
+        broadcast(:invalid)
+      end
+
+      protected
+
+      attr_reader :resource, :current_user
+
+      def invalid? = false
+
+      def destroy_resource
+        Decidim.traceability.perform_action!(
+          :delete,
+          resource,
+          current_user,
+          **extra_params
+        ) do
+          resource.destroy!
+        end
+      end
+
+      # Any extra params that you want to pass to the traceability service.
+      #
+      # @usage
+      #  def extra_params = { "visibility" => "all"}
+      #  def extra_params = { "visibility" => "public-only" }
+      #  def extra_params = { "visibility" => "admin-only" }
+      #  def extra_params = {
```

Comment:
I would only use the endless methods when it's oneliner. In this case I'd prefer to use just a boring method with an end

Reply:
changed in 94771d78819c3002354367f20f75b28d814444b9